app/drivers/usb.c

```diff
@@ -18,11 +15,28 @@ static struct {
 
 static uint8_t *usb_desc_device(USBD_SpeedTypeDef speed, uint16_t *p_len);
 static uint8_t *usb_desc_langid(USBD_SpeedTypeDef speed, uint16_t *p_len);
-static uint8_t *usb_desc_manuf_str(USBD_SpeedTypeDef speed, uint16_t *p_len);
-static uint8_t *usb_desc_product_str(USBD_SpeedTypeDef speed, uint16_t *p_len);
-static uint8_t *usb_desc_serial_str(USBD_SpeedTypeDef speed, uint16_t *p_len);
-static uint8_t *usb_desc_config_str(USBD_SpeedTypeDef speed, uint16_t *p_len);
-static uint8_t *usb_desc_if_str(USBD_SpeedTypeDef speed, uint16_t *p_len);
+static uint8_t *usb_desc_cfg(USBD_SpeedTypeDef speed, uint16_t *p_len);
+static uint8_t *usb_desc_device_qualifier(uint16_t *p_len);
+static uint8_t *usb_desc_usr_str(USBD_HandleTypeDef *p_dev, uint8_t idx, uint16_t *p_len);
+
+#define USBD_LANGID_STRING				1033
+#define USBD_VID						0x1209
+#define USBD_PID_FS						0x9450
+#define USBD_MANUFACTURER_STRING		"xil.se"
+#define USBD_PRODUCT_STRING_FS			"XilDebug CMSIS-DAP"
+#define USBD_SERIALNUMBER_STRING_FS		"123456789ABC"
+#define CDC_CTRL_INTERFACE_NO			0
+#define CDC_DATA_INTERFACE_NO			1
+#define HID_INTERFACE_NO				2
+
+#define USBD_USER_STRING_CDC			"CMSIS-DAP CDC"
+#define USBD_USER_STRING_CDC_IDX		4
+
+#define USBD_USER_STRING_DCI			"CMSIS-DAP DCI"
```

Comment:
What is DCI?

Reply:
Data Class Interface https://doc.micrium.com/display/DOC/CDC+Class+Overview